x/evm/keeper/statedb_test.go

```diff
@@ -383,6 +383,26 @@ func (suite *KeeperTestSuite) TestState() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestCommittedState() {
+	suite.SetupTest()
+
+	var key = common.BytesToHash([]byte("key"))
+	var value1 = common.BytesToHash([]byte("value1"))
+	var value2 = common.BytesToHash([]byte("value2"))
+
+	suite.app.EvmKeeper.SetState(suite.address, key, value1)
+
+	suite.app.EvmKeeper.BeginCachedContext()
+
+	suite.app.EvmKeeper.SetState(suite.address, key, value2)
+	tmp := suite.app.EvmKeeper.GetState(suite.address, key)
+	suite.Require().Equal(value2, tmp)
+	tmp = suite.app.EvmKeeper.GetCommittedState(suite.address, key)
+	suite.Require().Equal(value1, tmp)
+
+	suite.app.EvmKeeper.EndCachedContext()
```

Comment:
add test after the original context is recovered

Reply:
done, also added integration tests to check the gas consumption of sstore opcode, but we need to merge https://github.com/tharsis/ethermint/pull/348 first, otherwise, estimateGas will fail.

Reply:
I checked manually that the integration test fails on main branch(with fix-access-list branch merged):
```
AssertionError: expected '46466' to equal '43754'
```

Reply:
we need to `commit()` before recovering the context

Reply:
done